packages/server/wsHandler.ts

```diff
@@ -95,154 +98,177 @@ const dehydrateResult = (result: ExecutionResult) => {
-        subscribe
+      const args: EnvelopedExecutionArgs = {
+        schema: authToken.rol === 'su' ? privateSchema : schema,
+        operationName: params.operationName,
+        document,
+        variableValues: params.variables,
+        // dataLoader will be null for subscriptions
+        contextValue: {
+          dataLoader: extra.dataLoaders[id] || null,
+          rateLimiter,
+          ip,
+          authToken,
+          socketId
+        },
+        rootValue: {
+          execute,
+          subscribe
+        }
       }
+      return args
+    },
+    onComplete: (ctx, id) => {
+      const {extra} = ctx
+      const {dataLoaders} = extra
+      dataLoaders[id]?.dispose()
+      delete dataLoaders[id]
+    },
+    onDisconnect: async (ctx) => {
+      const {extra} = ctx
+      const {authToken, socketId, socket} = extra
+      socketTimers.delete(socket)
+      const {sub: userId} = authToken
+      Object.values(extra.dataLoaders).forEach((dl) => dl.dispose())
+      extra.dataLoaders = {}
+      activeClients.delete(extra.socketId)
+      const {execute, parse} = yoga.getEnveloped(ctx)
+      const dataLoader = getNewDataLoader('wsHandler.onDisconnect')
+      extra.socket.closed = true
+      await execute({
+        document: parse(disconnectQuery),
+        variableValues: {userId, socketId},
+        schema: privateSchema,
+        contextValue: {dataLoader, ip: extra.ip, authToken, socketId}
+      })
+      dataLoader.dispose()
     }
-    return args
   },
-  onComplete: (ctx, id) => {
-    const {extra} = ctx
-    const {dataLoaders} = extra
-    dataLoaders[id]?.dispose()
-    delete dataLoaders[id]
-  },
-  onDisconnect: async (ctx) => {
-    const {extra} = ctx
-    const {authToken, socketId} = extra
-    const {sub: userId} = authToken
-    Object.values(extra.dataLoaders).forEach((dl) => dl.dispose())
-    extra.dataLoaders = {}
-    activeClients.delete(extra.socketId)
-    const {execute, parse} = yoga.getEnveloped(ctx)
-    const dataLoader = getNewDataLoader('wsHandler.onDisconnect')
-    extra.socket.closed = true
-    await execute({
-      document: parse(disconnectQuery),
-      variableValues: {userId, socketId},
-      schema: privateSchema,
-      contextValue: {dataLoader, ip: extra.ip, authToken, socketId}
-    })
-    dataLoader.dispose()
+  {
+    message(socket) {
+      // graphql-ws doesn't expose the "WebSocket" they use that requires the onPing method
+      // So instead of refreshing Redis on ping, we do it every max 8 seconds
+      // That way we don't have to sniff what's in the message
+      const existing = socketTimers.get(socket)
+      if (existing) clearTimeout(existing)
+      const timer = setTimeout(() => {
+        socketTimers.delete(socket)
+        const {userId, closed} = socket as WebSocket<any> & UpgradeData
+        if (closed) return
+        getRedis().pexpire(`awareness:${userId}`, 10_000)
+      }, 8_000)
+      socketTimers.set(socket, timer)
+    }
```

Comment:
-1 This part seems wrong to me. We only refresh the key if we do not get a message within 8s. The ping is every 10s so that's why it most likely appears to work, but if there is more traffic on the socket, we'll never refresh the TTL.
I think this was meant to be some type of debounce? We have various debounce packages as transitive dependencies, we can just use one of these or ignore it altogether and just call redis each time.

The TTL should be increased to 12s or more. Or we get rid of it completely, see `User.isConnected`